homeassistant/components/history_stats/sensor.py

```diff
@@ -105,6 +106,13 @@ async def async_setup_platform(
     await coordinator.async_refresh()
     if not coordinator.last_update_success:
         raise PlatformNotReady from coordinator.last_exception
+
+    async def _on_hass_stop(_: Event) -> None:
+        """Shutdown coordinator on HomeAssistant stop."""
+        await coordinator.async_shutdown()
```

Comment:
Do we also need to consider reloads?
Should we implement `async_reset_platform`?
cc @bdraco